apps/client/src/components/Modals/NewChapterModal.svelte

```diff
@@ -26,15 +23,20 @@
           },
         },
       });
-      console.log('New lesson added successfully!');
-      document.getElementById('close')?.click();
+      console.log('New chapter added successfully!');
+      goto(location.href, { replaceState: true, noScroll: true });
```

Comment:
How does this affect the state? Does it maintain it or it gets lost? More importantly, is it achieving the desired behavior?

Reply:
It is supposed to Re-fetch the data from +page.ts/+layout.ts but currently, it is being called from a nested component. I think we need to lift these submit functions to the layout component if possible. Though am not sure if event forwarding can run that high up tho (button => modal => manage comp => layout)